connection.go

```diff
@@ -182,7 +182,15 @@ func (mc *mysqlConn) Prepare(query string) (driver.Stmt, error) {
 	columnCount, err := stmt.readPrepareResultPacket()
 	if err == nil {
 		if stmt.paramCount > 0 {
-			if err = mc.readUntilEOF(); err != nil {
+			// FIXME - seems like a bug in MySQL (or it's intended).
+			// There's no EOF return after parameters.
+			// However, this behavior isn't consistent to Maria DB.
+			if mc.flags&clientDeprecateEOF == 0 {
+				if err = mc.readUntilEOF(); err != nil {
+					return nil, err
+				}
+			}
```

Comment:
Seems MySQL behavior isn't consistent to the documentation here: https://dev.mysql.com/doc/internals/en/com-stmt-prepare-response.html

However, Maria DB follows this specification.

IMO this feature should be held until there's a clear solution to be compatible with supported vendors.

Reply:
https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_com_stmt_prepare.html#sect_protocol_com_stmt_prepare_response

Reply:
Can I continue your work? Does this problem be solved?